bitmark-references/bitmark-cli/bitmark-cli.md

```diff
@@ -0,0 +1,703 @@
+~~~
+
+
+## grant
+
+Transfer a number of shares of a particular id to a receiver.
+
+### OPTIONS
+
+-------------------------------  ---------------------------------
+`--receiver|-r IDENTITY|ACCOUNT` The identity or Base58 account that
+                                 is to receive the shares.
+
+`--share-id|-s SHARE-ID`         The transaction id of the share record,
+                                 which is used to identify the fractional item.
+
+`--quantity|-q NUMBER`           The quantity of share to grant to the receiving
+                                 account.  The default is one and can
+                                 be any value up to the current
+                                 balance of the owner.
+
+`--before-block|-b NUMBER`       This is used to provide a "time" limit on
+                                 the operation and the transaction
+                                 must be submitted and confirmed
+                                 before this block is present on the
+                                 blockchain.
+-------------------------------  ---------------------------------
+
+### EXAMPLE
+
+~~~
+bitmark-cli --identity=fred --network=testing grant --receiver=barney --share-id=8981eb58c965e2360b3ffeedf47d8f770a2fff7f67e5c348302d839afbf83dee --quantity=1000 --before-block=234765
+~~~
+
+
+## swap
+
+Perform an atomic swap between two different shares.  The owner send
+quantity-one share-one to the recipient.  The recipient sends
+quantity-two of share-two to the owner.  The system ensures the swap
+occurs as a single operation and no possibility of partial completion.
+This is a two signature command ad the recipient must use
+`countersign` to submit and pay before the expiry block is reached on
+the blockchain.
+
+### OPTIONS
+
+-------------------------------  ---------------------------------
+`--receiver|-r IDENTITY|ACCOUNT` The identity or Base58 account that
+                                 is to receive the shares.
+
+`--share-id-one|-s SHARE-ID`     The transaction id of the share record,
+                                 which is used to identify the
+                                 fractional item.
+
+`--quantity-one|-q NUMBER`       The quantity of share to grant to the
+                                 receiving account.  The default is
+                                 one and can be any value up to the
+                                 current balance of the owner.
+
+`--share-id-two|-S SHARE-ID`     The transaction id of the share record,
+                                 which is used to identify the
+                                 fractional item.
+
+`--quantity-two|-Q NUMBER`       The quantity of share to grant to the
+                                 receiving account.  The default is
+                                 one and can be any value up to the
+                                 current balance of the owner.
+
```

Comment:
Using the term "receiving account" for both quantity-one and quantity-two is confusing. We need to clearly differentiate the two different people in this transaction and who is sending which shares to whom. The "initiator" and the "countersigner" might be good terms. 

Thus, we'd have entries like:

-share-id-one: The transaction id of the share record that the initiator is sending, used to identify the first fractional item.

--quantity-one: The quantity of share-id-one that the initiator is granting to the countersigner. The default is one and can be any value up to the current balance of the initiator.

-share-id-two: The transaction id of the share record that the countersigner is sending, used to identify the first fractional item.

--quantity-two: The quantity of share-id-two that the countersigner is granting to the initiator. The default is one and can be any value up to the current balance of the initiator.